ray-operator/test/sampleyaml/support.go

```diff
@@ -74,3 +77,55 @@ func AllPodsRunningAndReady(pods []corev1.Pod) bool {
 	}
 	return true
 }
+
+func IsRayReady(t Test, pods []corev1.Pod) bool {
+	cmd := []string{
+		"python",
+		"-c",
+		"import ray; ray.init(); print(ray.cluster_resources())",
+	}
+
+	for _, pod := range pods {
+		req := t.Client().Core().CoreV1().RESTClient().
+			Post().
+			Resource("pods").
+			Name(pod.Name).
+			Namespace(pod.Namespace).
+			SubResource("exec").
+			VersionedParams(&corev1.PodExecOptions{
+				Command: cmd,
+				Stdin:   false,
+				Stdout:  true,
+				Stderr:  true,
+				TTY:     false,
+			}, clientgoscheme.ParameterCodec)
+
+		cfg := t.Client().Config()
+		exec, err := remotecommand.NewSPDYExecutor(&cfg, "POST", req.URL())
```

Comment:
I found a function called `ExecPodCmd` here

https://github.com/ray-project/kuberay/blob/1213d1578ac08a5fc993e2f1eb34fa021fe07749/ray-operator/test/support/core.go#L63-L71

We can use it directly in this PR.

However I think your use of client-go library is better, so I'll create an issue for tracking this and assign it to you if you are interested in it.

Reply:
tracking issue: https://github.com/ray-project/kuberay/issues/2462

Reply:
No problem